core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -296,33 +296,40 @@ KOKKOS_INLINE_FUNCTION auto mapping_from_ctor_and_8sizes(
     [[maybe_unused]] const size_t arg_N3, [[maybe_unused]] const size_t arg_N4,
     [[maybe_unused]] const size_t arg_N5, [[maybe_unused]] const size_t arg_N6,
     [[maybe_unused]] const size_t arg_N7) {
+  const size_t N0 = arg_N0 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N0;
+  const size_t N1 = arg_N1 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N1;
+  const size_t N2 = arg_N2 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N2;
+  const size_t N3 = arg_N3 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N3;
+  const size_t N4 = arg_N4 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N4;
+  const size_t N5 = arg_N5 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N5;
+  const size_t N6 = arg_N6 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N6;
+  const size_t N7 = arg_N7 == KOKKOS_IMPL_CTOR_DEFAULT_ARG ? 1 : arg_N7;
```

Comment:
This is probably worth a comment (in the code)
In what circumstances we get the default argument passed?

Reply:
if we don't pass enough arguments (less thank rank) to a View constructor. We expect to detect that on the Kokkos-side but didn't get there since `mdspan` just reported an overflow in `span` calculations.

Reply:
I kind of prefer it in `mapping_from_ctor_and_8sizes` because that acts as a translation between the Kokkos API and mdspan AI